src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +255,33 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || isPrivateWithDefaultCtor(desc))
                 && !(desc.isDeclaredAsAbstract()
                     || desc.isSuperClassOfAnonymousInnerClass())
                 && !desc.isDeclaredAsFinal()
                 && !desc.isWithNonPrivateCtor()
                 && !desc.isWithNestedSubclass();
     }
 
+    /**
+     * Check if class is private and with default constructor.
+     *
+     * @param desc description of class
+     * @return true if class is private and with default constructor
+     */
+    private static boolean isPrivateWithDefaultCtor(ClassDesc desc) {
+        final DetailAST descAst = desc.getTypeDeclarationAst();
+        DetailAST modifierChild = descAst.getFirstChild().getFirstChild();
+        final List<Integer> list = new ArrayList<>();
```

Comment:
Why do you need to create a list and sotre it, when we are only looking for `LITERAL_PRIVATE`? Switch this to `findFirstToken`.

The `ClassDesc` constructor also looks at the modifiers and stores them into fields. You should follow it's pattern.

Reply:
not done yet

Reply:
@Kevin222004 Is there an issue for this?

Reply:
done

Reply:
ping